src/d/actor/d_a_obj_vfan.cpp

```diff
@@ -4,54 +4,258 @@
+
+    m_evid = dComIfGp_evmng_getEventIdx("Vfan", 0xff);
+
+    return TRUE;
 }
 
 /* 00000214-000003D0       .text Mthd_Create__Q29daObjVfan5Act_cFv */
-void daObjVfan::Act_c::Mthd_Create() {
-    /* Nonmatching */
+cPhs__Step daObjVfan::Act_c::Mthd_Create() {
+    fopAcM_SetupActor(this, daObjVfan::Act_c);
+
+    cPhs__Step phase_state;
+
+    if (fopAcM_isSwitch(this, prm_get_swSave())) {
+        return cPhs_UNK3_e;
+    } else {
+        phase_state = (cPhs__Step)dComIfG_resLoad(&mPhs, M_arcname);
+        if (phase_state == cPhs_COMPLEATE_e) {
+            phase_state = (cPhs__Step)MoveBGCreate(M_arcname, 7, NULL, 0xa60);
+
+            JUT_ASSERT(0xc6, (phase_state == cPhs_COMPLEATE_e) || (phase_state == cPhs_ERROR_e));
+        }
+    }
+
+    return phase_state;
 }
 
 /* 00000588-00000590       .text Delete__Q29daObjVfan5Act_cFv */
 BOOL daObjVfan::Act_c::Delete() {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00000590-000005E8       .text Mthd_Delete__Q29daObjVfan5Act_cFv */
-void daObjVfan::Act_c::Mthd_Delete() {
-    /* Nonmatching */
+BOOL daObjVfan::Act_c::Mthd_Delete() {
+    int res = MoveBGDelete();
+    if (mBase.mCreateResult != 3) {
+        dComIfG_resDelete(&mPhs, M_arcname);
+    }
+    return res;
 }
 
 /* 000005E8-00000680       .text set_mtx__Q29daObjVfan5Act_cFv */
 void daObjVfan::Act_c::set_mtx() {
-    /* Nonmatching */
+    shape_angle = current.angle;
+
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::ZXYrotM(shape_angle);
+
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
+    MTXCopy(mDoMtx_stack_c::get(), M_tmp_mtx);
 }
 
 /* 00000680-000006BC       .text init_mtx__Q29daObjVfan5Act_cFv */
 void daObjVfan::Act_c::init_mtx() {
-    /* Nonmatching */
+    mpModel->mBaseScale = mScale;
+
+    set_mtx();
 }
 
 /* 000006BC-000009B4       .text ParticleSet__Q29daObjVfan5Act_cFv */
 void daObjVfan::Act_c::ParticleSet() {
-    /* Nonmatching */
+    // was this manually unrolled?
+    dComIfGp_particle_set(0x83cd, &current.pos, &current.angle);
+    dComIfGp_particle_set(0x83ce, &current.pos, &current.angle);
+    dComIfGp_particle_set(0x83cf, &current.pos, &current.angle);
+    dComIfGp_particle_set(0x83d0, &current.pos, &current.angle);
+    dComIfGp_particle_set(0x83d1, &current.pos, &current.angle);
+    dComIfGp_particle_set(0x83d2, &current.pos, &current.angle);
+    dComIfGp_particle_set(0x83d3, &current.pos, &current.angle);
+    dComIfGp_particle_set(0x83d4, &current.pos, &current.angle);
+    dComIfGp_particle_set(0x83d5, &current.pos, &current.angle);
+    dComIfGp_particle_set(0x83d6, &current.pos, &current.angle);
+    dComIfGp_particle_set(0x83d7, &current.pos, &current.angle);
 }
 
 /* 000009B4-00000C74       .text Execute__Q29daObjVfan5Act_cFPPA3_A4_f */
-int daObjVfan::Act_c::Execute(float(**)[3][4]) {
-    /* Nonmatching */
+int daObjVfan::Act_c::Execute(Mtx** mtx) {
+    switch (mSwitchNo) {
+    case 0:
+        g_dComIfG_gameInfo.play.mCcS.Set(static_cast<cCcD_Obj*>(&mCyl));
```

Comment:
```suggestion
        dComIfG_Ccsp()->Set(&mCyl);
```
manual cast to base class not required